consensus/core/src/transaction.rs

```diff
@@ -370,15 +398,18 @@ impl TransactionClient {
             }
         }
 
+        let n = transactions.len();
         let t = TransactionsGuard {
             transactions: transactions.into_iter().map(Transaction::new).collect(),
             included_in_block_ack: included_in_block_ack_send,
         };
         self.sender
             .send(t)
             .await
-            .tap_err(|e| error!("Submit transactions failed with {:?}", e))
+            .tap_err(|e| error!("Submit transactions failed with {}", e))
             .map_err(|e| ClientError::ConsensusShuttingDown(e.to_string()))?;
+        self.inflight.fetch_add(n, Ordering::AcqRel);
```

Comment:
I am not 100% sure of the cancel safety capabilities of the `sender.send.await` here, but I am thinking that we could face some race conditions in case of future cancellation (the value is sent to the underlying channel but the future is cancelled and the `inflight` never gets incremented).

Maybe the following will give us stronger guarantees?

```
      let permit = self.sender.reserve().await
           .tap_err(|e| error!("Submit transactions failed with {}", e))
            .map_err(|e| ClientError::ConsensusShuttingDown(e.to_string()))?;
      permit.send(t);
      
     self.inflight.fetch_add(n, Ordering::AcqRel);
```